src/Adapter/MSTestAdapter.PlatformServices/Execution/UnitTestRunner.cs

```diff
@@ -166,7 +172,7 @@ internal async Task<TestResult[]> RunSingleTestAsync(TestMethod testMethod, IDic
                 _assemblyFixtureTests.TryAdd(testMethod.AssemblyName, testMethodInfo.Parent.Parent);
                 _classFixtureTests.TryAdd(testMethod.AssemblyName + testMethod.FullClassName, testMethodInfo.Parent);
 
-                ITestContext testContextForAssemblyInit = PlatformServiceProvider.Instance.GetTestContext(testMethod, writer, properties, messageLogger, testContextForTestExecution.Context.CurrentTestOutcome);
+                testContextForAssemblyInit = PlatformServiceProvider.Instance.GetTestContext(testMethod, writer, properties, messageLogger, testContextForTestExecution.Context.CurrentTestOutcome);
```

Comment:
Can't we dispose that earlier? Same question for the various instances.

Reply:
We can but it's annoying to have too many `try/finally`. I think it's not very problematic to dispose a bit later.